firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -106,15 +125,27 @@ internal class SessionLifecycleService : Service() {
    */
   private fun handleForegrounding(msg: Message) {
     Log.i(TAG, "Activity foregrounding at ${msg.getWhen()}")
-
     if (!hasForegrounded) {
       Log.i(TAG, "Cold start detected.")
       hasForegrounded = true
       broadcastSession()
+      updateSessionStorage(SessionGenerator.instance.currentSession.sessionId, msg.getWhen())
     } else if (msg.getWhen() - lastMsgTimeMs > MAX_BACKGROUND_MS) {
       Log.i(TAG, "Session too long in background. Creating new session.")
       SessionGenerator.instance.generateNewSession()
       broadcastSession()
+      updateSessionStorage(SessionGenerator.instance.currentSession.sessionId, msg.getWhen())
+    }
+  }
+
+  private fun updateSessionStorage(timeMs: Long) {
+    updateSessionStorage(null, timeMs)
+  }
+
+  private fun updateSessionStorage(sessionId: String?, timeMs: Long) {
```

Comment:
You could say `sessionId: String = null` here instead of the manual overloading with and without sessionId.